Document/0x05a-Platform-Overview.md

```diff
@@ -12,23 +12,35 @@ Visit the official [Android developer documentation website](https://developer.a
+
+Dalvik bytecode is an optimized version of Java bytecode. It is created by first compiling the Java or Kotlin code to Java bytecode, using the javac and kotlinc compilers respectively, producing .class files. Finally, the Java bytecode is converted to Dalvik bytecode using the d8 tool. Dalvik bytecode is packed within APK and AAB files in the form of .dex files and is used by a managed runtime on Android to execute it on the device.
 
 ![OWASP MSTG](Images/Chapters/0x05a/java_vs_dalvik.png) \
 
-The current version of Android executes this bytecode on the Android runtime (ART). ART is the successor to Android's original runtime, the Dalvik Virtual Machine (DVM). The key difference between Dalvik and ART is the way the bytecode is executed.
+Before Android 5.0 (API level 21), Android executed bytecode on the Dalvik Virtual Machine (DVM), where it was translated into machine code at execution time, a process known as *just-in-time* (JIT) compilation. This enables the runtime to benefit from the speed of compiled code while maintaining the flexibility of code interpretation.
+
+Since Android 5.0 (API level 21), Android executes bytecode on the [Android Runtime (ART)](https://source.android.com/devices/tech/dalvik/configure#how_art_works) which is the successor of the DVM. ART provides improved performance as well as context information in app native crash reports, by including both Java and native stack information. It uses the same Dalvik bytecode input to maintain backward compatibility. However, ART executes the Dalvik bytecode differently, using a hybrid combination of *ahead-of-time* (AOT), JIT and profile-guided compilation.
+
+- **AOT** pre-compiles Dalvik bytecode into native code, and the generated code will be saved on disk with the .oat extension (ELF binary). dex2oat is a tool used to perform the compilation and can be found at /system/bin/dex2oat on Android devices. AOT during the installation, or when the device is not in use, but this approach has a downside to prolonged application install time and operating system update time, due to it is necessary to recompile all applications.
+- **JIT** happens at runtime.
+- **Profile-guided compilation** is a hybrid approach that was introduced in recent versions of the ART runtime to combat the downsides of AOT. After installation, all parts of the application will be interpreted and methods frequently executed will be JIT compiled, at the same time profiles getting generated to trace the frequently used code part. When the device is idle, a compilation (dex2oat) daemon runs, and AOT compiles code based on the generated profile from previous runs.
+
+Apps are recompiled on the device when they are installed, or when the OS undergoes a major update. While the code is being recompiled, device-specific and advanced code optimizations techniques are applied. The final recompiled code is then used for all subsequent executions.

```

Comment:
```suggestion
```
I suggest to delete this. This info is now in the AOT section because it made sense to mention there why the system update takes so long.
(I created my suggestion before seeing this paragraph 😬 )